src/alg_utils.jl

```diff
@@ -49,11 +49,35 @@ get_current_isfsal(alg::CompositeAlgorithm, cache) = isfsal(alg.algs[cache.curre
 issplit(alg::Union{OrdinaryDiffEqAlgorithm,DAEAlgorithm}) = false
 issplit(alg::SplitAlgorithms) = true
 
+@generated function _composite_beta1_default(algs::T, current, ::Type{QT}) where {T <: Tuple, QT}
+  expr = Expr(:block)
+  for i in 1:length(T.types)
+    push!(expr.args, quote
+      if current == $i
+        return QT(beta1_default(algs[$i], beta2_default(algs[1])))
+      end
+    end)
+  end
+  return expr
+end
+@generated function _composite_beta2_default(algs::T, current, ::Type{QT}) where {T <: Tuple, QT}
+  expr = Expr(:block)
+  for i in 1:length(T.types)
+    push!(expr.args, quote
+      if current == $i
+        return QT(beta2_default(algs[1]))
```

Comment:
This seems incorrect:
```suggestion
        return QT(beta2_default(algs[$i]))
```

Reply:
oh dear :facepalm: